test/scroll_list/ScrollListSpec.js

```diff
@@ -299,25 +298,138 @@ define(function(require) {
+                expect(firstItem.left).toBe(expectedLeft);
+                expect(firstItem.right).toBe(expectedRight);
+                expect(secondItem.scale).toBe(zoomToScale);
+                expect(secondItem.itemIndex).toBe(1);
+            });
+
+            it('should filter out items that are not in the viewport once accounting for padding', function() {
+                var options = {
+                    itemCount: 2,
+                    padding: 10,
+                    gap: 10
+                };
+                // Ensure that the items fill the viewport, and the first item
+                // with padding and gap fills the viewport completely.
+                options.itemWidth = VIEWPORT_WIDTH - options.padding * 2;
+                options.itemHeight = VIEWPORT_HEIGHT - options.padding * 2;
+                // Zoom and scroll such that there is padding/gap from the first
+                // item visible.
+                options.zoomToScale = 2;
+                options.scrollToX = undefined;
+                options.scrollToY = VIEWPORT_HEIGHT - options.gap / 2;
+
+                var scrollList = setupTest(options);
+                var itemsInViewport = scrollList.getVisibleItemPositionData();
+                expect(itemsInViewport.length).toBe(1);
             });
-            it('should call hit tester with item map if not in "flow" mode', function() {
-                var fakeResult = {};
-                spyOn(HitTester, 'testItemMap').andReturn(fakeResult);
-                testScrollList({ mode: '!flow' }, function(scrollList) {
-                    var point = { x: 1, y: 2 };
-                    var result = scrollList.hitTest(point);
-                    expect(HitTester.testItemMap).toHaveBeenCalledWith(scrollList, point);
-                    expect(result).toBe(fakeResult);
+        });
+
+        describe('hit testing', function() {
+            it('should get visible item positions and hit test against them', function() {
+                testScrollList(function(scrollList) {
+                    spyOn(scrollList, 'getVisibleItemPositionData').andReturn([
+                        { itemIndex: 0, top: 10, bottom: 100, left: 20, right: 200, scale: 2 },
+                        { itemIndex: 1, top: 110, bottom: 200, left: 20, right: 200, scale: 2 }
+                    ]);
+
+                    // Expect false when nothing is hit
+                    var hitResult = scrollList.hitTest({ x: 0, y: 0 });
+                    expect(hitResult).toBe(false);
+
+                    // Expect position relative to actual item size for each item in view.
+                    hitResult = scrollList.hitTest({ x: 100, y: 50 });
+                    expect(hitResult).toEqual({ index: 0, position: { x: 40, y: 20 }});
+
+                    hitResult = scrollList.hitTest({ x: 50, y: 120 });
+                    expect(hitResult).toEqual({ index: 1, position: { x: 15, y: 5 }});
```

Comment:
I can't see the relationships with the sizes (relative positions). Could you make the position calculations explicit? Or at least commented.